src/foam/nanos/auth/CreatedAwareDAO.js

```diff
@@ -19,11 +19,11 @@ foam.CLASS({
   methods: [
     {
       name: 'put_',
-      code: function(value) {
-        if ( foam.util.SafetyUtil.isEmpty(obj.id) ) {
-          value.created = new Date();
+      code: function(x, obj) {
+        if ( ! obj.id ) {
+          obj.created = new Date();
```

Comment:
I feel like the JS should also check that obj implements CreatedAware same as Java.

Reply:
In java we check if delegate.find_(x, id) == null, can we do the same for JS?  It would be best to keep these implementations consistent.

I also don't like checks for if ( obj.id ), because that makes values like 0 not a valid id but sometimes people try to use that, let's just check delegate.find_() or just check obj.hasOwnProperty('id') like SequenceNumberDAO does.